File: backend/collection/nest/http.go
package main

import (
	"bytes"
	"encoding/json"
	"errors"
	"io"
	"net/http"
)

func sendEmptyPost(url string) ([]byte, error) {
	postBody, _ := json.Marshal(map[string]string{})
	responseBody := bytes.NewBuffer(postBody)

	resp, err := http.Post(url, "application/json", responseBody)
	if err != nil {
		return nil, errors.New("Failed to send POST.")
	}
	defer resp.Body.Close()

	body, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, errors.New("Failed to read response body.")
	}

	return body, nil
}

func sendGetRequestWithAccessToken(url, accessToken string) ([]byte, error) {
	// Empty body.  Header has to be set after the request is created.
	req, err := http.NewRequest("GET", url, nil)
	if err != nil {
		return nil, errors.New("Failed to initialize request.")
	}

	req.Header.Set("Content-Type", "application/json")
	req.Header.Set("Authorization", "Bearer "+accessToken)
	client := &http.Client{}

	resp, err := client.Do(req)
	if err != nil {
		return nil, errors.New("Failed to send GET.")
	}
	defer resp.Body.Close()

	body, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, errors.New("Failed to read response body.")
	}

	return body, nil
}

File: backend/ingest/nest/http.py
from typing import Any
import requests


def sendEmptyPost(url: str) -> dict[str, Any]:
    response = requests.post(
        url=url,
        data={},
        headers={"Content-type": "application/json"},
    )

    if response.status_code != 200:
        raise requests.HTTPError(f"Empty POST failed with status code {response.status_code}.")
    else:
        return response.json()


def sendGetRequestWithAccessToken(url: str, accessToken: str) -> dict[str, Any]:
    response = requests.get(
        url=url,
        headers={
            "Content-type": "application/json",
            "Authorization": f"Bearer {accessToken}",
        },
    )

    if response.status_code != 200:
        raise requests.HTTPError(f"GET failed with status code {response.status_code}")
    else:
        return response.json()
